migrations/2018_01_18_132900_create_notifications_from_table.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+return [
+    'up' => function (Builder $schema) {
+        $schema->create('notifications_from', function (Blueprint $table) {
+            $table->integer('id')->unsigned();
+            $table->integer('from_user_id')->unsigned();
+
+            $table->foreign('id')->references('id')->on('notifications')->onDelete('cascade');
+            $table->foreign('from_user_id')->references('id')->on('users')->onDelete('cascade');
+        });
+
+        $schema->getConnection()->table('notifications')->chunkById(100, function ($notifications) use ($schema) {
+            foreach ($notifications as $notification) {
+                $insert = [
+                    'id' => $notification->id,
+                    'from_user_id' => $notification->sender_id
+                ];
+
+                $schema->getConnection()->table('notifications_from')->updateOrInsert($insert, $insert);
+            }
+        });
```

Comment:
> we don't add multiple table statements into one migration because migrations could be half run during an exception

Does this migration break that rule? Should the insertion part be in a separate migration?

Reply:
Perhaps the insertion part should be in the same migration as where the `sender_id` column is dropped

Reply:
Yes, this won't work; if the update or insert call throws an exception the migration will not be marked done, but re-running the migration isn't possible either because the first part already finished..

Reply:
Fixed in https://github.com/flarum/core/pull/1344/commits/1e372f3881aebb6aa930b53201067e593684607c.